addons/test_mail/tests/test_mail_alias.py

```diff
@@ -386,6 +386,58 @@ def test_alias_setup(self):
 class TestAliasCompany(TestMailAliasCommon):
     """ Test company / alias domain and configuration synchronization """
 
+    def test_alias_domain_setup_archived_company(self):
+        """Test initialization of alias domain with at least one archived company
+        and at least one mail.alias record points to one mail.thread of the
+        archived company"""
+
+        # add archived company to multi company setup
+        self.company_archived = self.env['res.company'].create({
+                'country_id': self.env.ref('base.be').id,
+                'currency_id': self.env.ref('base.EUR').id,
+                'email': 'company_archived@test.example.com',
+                'name': 'Company Archived Test',
+            })
+        self.company_archived.action_archive()
+
+        # create record inheriting from mail.thread to be used as owner/target thread
+        test_record_archived_company = self.env['mail.test.simple.unfollow'].create({
+                'name': 'Test record (mail.thread) specific to archived company',
+                'company_id': self.company_archived.id,
+            })
+
+        unfollow_model_id = self.env['ir.model']._get('mail.test.simple.unfollow').id
+        mc_archived_parent = self.env['mail.alias'].create({
+                'alias_name': 'alias_parent_specific_to_archived_company',
+                'alias_parent_model_id': unfollow_model_id,
+                'alias_model_id': unfollow_model_id,
+                'alias_parent_thread_id': test_record_archived_company.id,
+            })  # case where the parent thread is specific to archived company
+
+        mc_archived_target = self.env['mail.alias'].create({
+                'alias_name': 'alias_target_specific_to_archived_company',
+                'alias_parent_model_id': unfollow_model_id,
+                'alias_model_id': unfollow_model_id,
+                'alias_force_thread_id': test_record_archived_company.id,
+            })  # case where the target thread is specific to archived company
+
+        # eject linked aliases then remove all alias domains; should
+        # trigger the init condition at next create() call
+        self.env['mail.alias'].search([]).write({'alias_domain_id': False})
+        self.env['mail.alias.domain'].search([]).unlink()
```

Comment:
Could check we effectively have a void domain on test records, just to be sure of test conditions